streaming/text/enwiki.py

```diff
@@ -0,0 +1,101 @@
+
+from transformers.models.auto.tokenization_auto import AutoTokenizer
+
+from streaming.base import Dataset
+
+
+class EnWiki(Dataset):
+    """Implementation of the English Wikipedia 2020-01-01 streaming dataset.
+
+    Args:
+        tokenizer_name (str): The name of the HuggingFace tokenizer to use to tokenize samples.
+        max_seq_len (int): The max sequence length of each token sample.
+        group_method (str): How to group text samples into token samples. Currently only supporting
+            ``'truncate'``.
+        local (str): Local dataset directory where shards are cached by split.
+        remote (str, optional): Download shards from this remote path or directory. If None, this
+            rank and worker's partition of the dataset must all exist locally. Defaults to
+            ``None``.
+        split (str, optional): Which dataset split to use, if any. Defaults to ``None``.
+        shuffle (bool): Whether to shuffle the samples while iterating. Defaults to ``True``.
+        prefetch (int, optional): Target number of samples remaining to prefetch while iterating.
+            Defaults to ``None``.
+        keep_zip (bool, optional): Whether to keep or delete the compressed file when
+            decompressing downloaded shards. If set to None, keep iff remote is local. Defaults to
+            ``None``.
+        retry (int): Number of download re-attempts before giving up. Defaults to ``2``.
+        timeout (float): Number of seconds to wait for a shard to download before raising an
+            exception. Defaults to ``60``.
+        hash (str, optional): Optional hash or checksum algorithm to use to validate shards.
+            Defaults to ``None``.
+        batch_size (int, optional): Hint the batch_size that will be used on each device's
+            DataLoader. Defaults to ``None``.
+    """
+
+    def __init__(self,
+                 tokenizer_name: str,
+                 max_seq_len: int,
+                 group_method: str,
+                 local: str,
+                 remote: Optional[str] = None,
+                 split: Optional[str] = None,
+                 shuffle: bool = True,
+                 prefetch: Optional[int] = 100_000,
+                 keep_zip: Optional[bool] = None,
+                 retry: int = 2,
+                 timeout: float = 60,
+                 hash: Optional[str] = None,
+                 batch_size: Optional[int] = None) -> None:
+        if group_method not in ['truncate']:
+            raise ValueError(f'Only group_method="truncate" is supported at this time.')
+
+        super().__init__(local, remote, split, shuffle, prefetch, keep_zip, retry, timeout, hash,
+                         batch_size)
+        self.tokenizer_name = tokenizer_name
+        self.max_seq_len = max_seq_len
+        self.group_method = group_method
+
+        # Build tokenizer
+        self.tokenizer = AutoTokenizer.from_pretrained(self.tokenizer_name)  # pyright: ignore
```

Comment:
is this `# pyright: ignore` still relevant here of copy/paste ?

Reply:
Will try removing it

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/23/commits/926d9779baed22f1e9392a6b96b43cfc3f63d68e